content/docs/user-guide/basic-concepts/workspace.md

```diff
@@ -6,3 +6,46 @@ tooltip: >-
   models, etc. Typically, it's also a Git repository. It will contain your DVC
   project.
 ---
+
+<!-- keywords: data science project architecture, machine learning project architecture, machine learning workflow, data science workflow, machine learning file system, data science file system, data science project structure, machine learning project structure, notebook version control -->
+
+# Workspace
+
+A data science project consists of data obtained from many different sources.
+Most of the time it needs to convert the the format of this data into a form
+that is required by the training models and supplying to data science / machine
+learning workflows. Sometimes it needs to be split into multiple files or
+directories or (as the project structure needs) have different versions for
+different requirements. e.g. A smaller / simplified version might be required in
+protoyping for faster feedback and shorter training times.
+
+These activities all involve to change the machine learning / data science
+project structure in unforeseeable ways and tracking them manually becomes
+tedious. A single workspace to manage all artifacts of a project is desirable,
+although versioning needs and managing dependencies make it increasingly
+complex.
+
+DVC allows a single directory to contain all your project artifacts. In the
+documentation the workspace is the _user visible_ part of the directory that
+contains all your project files e.g. raw datasets, source code, ML models, etc.
+Users work in this directory using their data and model files that and
+manipulate the contents through DVC commands.
+
+Files and directories in the workspace can be added to DVC (`dvc add`) or they
+can be downloaded from external sources (`dvc get`, `dvc import`,
+`dvc import-url`). Changes to the files, directories, notebooks, models and in
+general machine learning file system can be tracked (`dvc commit`).
+<abbr>Pipelines</abbr> and <abbr>dependencies</abbr> between them can be defined
+(`dvc run`). Data and model files can be sent to or retrieved from
+<abbr>remotes</abbr> (`dvc push`, `dvc pull`) and can be removed (`dvc remove`,
+`dvc gc`) from the workspace. DVC supports all typical operations of files and
```

Comment:
Good idea for the secondary part of concept pages: mention/link to the most relevant commands and features. In this case I agree about `add/import*/remove/move`, and `commit`, but importantly, we're currently missing `checkout`. Maybe `push/pull` (not sure, maybe `checkout` is enough as it relates workspace and cache). Probably not `gc` or `run`.

Reply:
> cleanup operations are also a major hurdle for the common ML engineer

Sure but `gc` doesn't operate on the workspace, but rather on the cache (or remote). The workspace is always kept clean automatically (you only see 1 version at a time).

Def OK on `remove`, in fact it should probably be closer to the top (:

Similarly `push/pull` don't relate that much to the workspace, they're mostly operations between cache and remote. OK `pull` a little bit but it could be a parenthesis next to `checkout` probably.

Reply:
I'm also not seeing much need for the `remote` link BTW.

Reply:
This is addressed in bdbcc97